### ISSUE TITLE
(question) deployment issue to heroku, huge binary

### ISSUE THREAD

**Issue by u007** [gobuffalo/buffalo#378]
hi,

is there any reason not to have physical file on heroku that the binary can load after deployed?
why does build need to compile all assets into binary and serve it from the binary?

how do i add my own files to the directory that is configuration related? like language file...

**Comment by stanislas-m** [comment]
Hello,

> is there any reason not to have physical file on heroku that the binary can load after deployed?
why does build need to compile all assets into binary and serve it from the binary?

You can use the `-e` option to handle the assets manually (see: http://gobuffalo.io/docs/building#extract-assets). Did you mean you want the assets to be served by buffalo, but not stored in the binary?

**Comment by u007** [comment]
hi, i think i solved my issue.

i wanted to add a yaml file that is loaded when the app start.
since ive deployed to heroku, ive to hardcode the path to 
```
/app
```
with relative to this path, my configuration is loaded correctly.
except now that database.yml as ive read is build into the binary

**Comment by u007** [comment]
hi @stanislas-m , how do i manage assets manually?
binary seems to be too huge and it take a long time to deploy to heroku
do you have instruction on how to do it manually?

thanks

**Comment by u007** [comment]
hi Sorry, i find deployment is quite a pain to heroku,
anyway to speed up or do incremental git push instead of 1 binary that server all and 1 assets?
i realize major issue is the generated binary which size up to 40+mb
assets standalone zipped is only 4mb, how does it work?
anyway to make incremental binary?

**Comment by stanislas-m** [comment]
@u007 

> i realize major issue is the generated binary which size up to 40+mb
> assets standalone zipped is only 4mb, how does it work?

The way the extract option work is to prevent the assets to be embedded in the binary ; so you'll have the binary on one side (lighter) and the assets on the other side (probably heavier).

What's the size of the binary, with the extract option? (compared to no extract) Maybe there is a bug, since we changed the way to handle the embedding, or it's just the binary which is a bit heavy.

**Comment by u007** [comment]
sorry its 

- 25.2mb for bin/heroku without assets
- 4.6mb for assets.zip

and if its without -e option:

36+mb bin/heroku if inclusive assets

using v0.8.2

**Comment by stanislas-m** [comment]
Hum, so it seems the assets are really not included in the binary, so it still works. However, 25MB seems pretty huge, I'd take a look after work.

**Comment by stanislas-m** [comment]
@u007 maybe you can try to use the strip tags `--ldflags="-s -w"`, so the binary will be lighter.

**Comment by u007** [comment]
nice its 17+mb now :)

**Comment by u007** [comment]
when -e option is given, the templates it tried to open from my local development path when i compile it.
eg:
/Users/username/gopath/src/project/templates/admin/dashboard.html

**Comment by markbates** [comment]
@u007 that sounds like a bug. templates and migrations should always been in the binary, but if the `-e` flag is used then assets are the only things that shouldn't be bundled.

**Comment by u007** [comment]
by the way, if this assets is fixed, it will be a huge thumbs up. the memory consumed without the assets compiled into the binary is way lower. 20+mb vs 80+mb in my use case.  thank you 👍

**Comment by u007** [comment]
hi @markbates  
is it fixed? :)

**Comment by markbates** [comment]
It’s fixed in the development branch. Will be version 0.9.5 when released.

**Comment by ryanerwin** [comment]
> @u007 maybe you can try to use the strip tags `--ldflags="-s -w"`, so the binary will be lighter.

Is there a downside to stripping by default when building? And is there a way to pass default build flags that's part of gobuffalo, or just use our shell environment for that?

Are the same debugging symbols used by `delve` and `gdb`?
Fortunately I haven't had to debug golang in production yet, so I'm not sure of the consequences here...

I was just experimenting with porting an existing project over to gobuffalo for easier deployment, but I noticed the default binary was pretty large (24MB) but when you strip it, size drops to about 17MB. Not bad when you consider a golang "hello world" binary is 2MB. Also, I got a smaller binary running `ldflags=-s -w` (17MB) vs 19MB when I just ran `strip` on the binary...
